### PR TITLE
Bug mesh io iterator data

### PR DESCRIPTION
**Pull request by roigcarlo**
This PR fixes should fix https://github.com/simphony/simphony-common/issues/89, a problem reported by @khiltunen which was making the data of the entities of the filemesh not being returned if the iter_xxxx() were called. 

TODO:
- [x] Travis pass

Once travis passes I will ping you for the merge.

### CODE REVIEW

**Comment by roigcarlo** [comment]
@khiltunen, please confirm that everything is ok so we can merge

File: simphony/io/tests/test_file_mesh.py [hunk 1]
```diff
@@ -375,6 +375,93 @@ def test_get_all_cells_iterator(self):
...
+
+        euids = [self.mesh.add_edge(edge) for edge in edges]
+
+        iedges = self.mesh.iter_edges()
+
+        iedges_id = [edge.uid for edge in iedges]
+        first_edge = self.mesh.iter_edges().next()
+
+        self.assertItemsEqual(iedges_id, euids)
+        self.assertIsNot(len(first_edge.data), 0)
```
Comment (itziakos) [inline simphony/io/tests/test_file_mesh.py]: The `compare_data_containers` function can be used to setup an equality assertion for the TestCase (see usage in https://github.com/simphony/simphony-common/blob/master/simphony/testing/abc_check_particle_containers.py#L66)

Then one can just do 
```python

self.assertEqual(first_edge.data, DataContainer(VELOCITY=(0, 0, 0)))

File: simphony/io/tests/test_file_mesh.py [hunk 2]
```diff
@@ -375,6 +375,93 @@ def test_get_all_cells_iterator(self):
...
+        the edges of the mesh is returned
+        when the function iter_edges is called
+        without arguments and its data is correctly retrieved
+
+        """
+
+        puids = [self.mesh.add_point(point) for point in self.points[:3]]
+
+        edges = [
+            Edge(puids[0:2], data=DataContainer({CUBA.VELOCITY: [0, 0, 0]}))
```
Comment (itziakos) [inline simphony/io/tests/test_file_mesh.py]: I would suggest to use immutable objects like tuples for the `DataContainer` values. Especially tuples are faster to create and take much less memory resources than lists.

File: simphony/io/tests/test_file_mesh.py (hunk 1 above)
Comment (itziakos) [inline simphony/io/tests/test_file_mesh.py]: the correct link is https://github.com/simphony/simphony-common/blob/master/simphony/testing/abc_container_check.py#L15

**Comment by khiltunen** [comment]
It is OK on my side to merge. Fix made things work.

**Comment by nathanfranklin** [comment]
Looks good.  

Can a `test_get_all_points_iterator_with_data` also be written?

**Comment by itziakos** [comment]
:+1:

**Comment by nathanfranklin** [comment]
:+1:

**Pull request merged**